example/snake/main.cpp

```diff
@@ -0,0 +1,81 @@
+//
+// lager - library for functional interactive c++ programs
+// Copyright (C) 2019 Carl Bussey
+//
+// This file is part of lager.
+//
+// lager is free software: you can redistribute it and/or modify
+// it under the terms of the MIT License, as detailed in the LICENSE
+// file located at the root of this source code distribution,
+// or here: <https://github.com/arximboldi/lager/blob/master/LICENSE>
+//
+
+#include "actions.hpp"
+#include "model.hpp"
+#include "qmodel.hpp"
+#include "reduce.hpp"
+
+#include <lager/event_loop/qapplication.hpp>
+#include <lager/store.hpp>
+
+#include <QApplication>
+#include <QQmlApplicationEngine>
+#include <QQmlContext>
+#include <QTimer>
+
+#include <cmath>
+
+using namespace sn;
+
+constexpr int initial_update_period{100};
+constexpr int min_update_period{20};
+constexpr int update_period_decrement{10};
+
+int main(int argc, char** argv)
+{
+    QApplication app{argc, argv};
+    QQmlApplicationEngine engine;
+
+    auto initial_state            = make_initial();
+    const auto initial_snake_size = initial_state.snake.body.size();
+    auto store =
+        lager::make_store<action_t>(std::move(initial_state),
+                                    reduce,
+                                    lager::with_qapplication_event_loop{app});
+
+    auto* timer = new QTimer(&app);
+    QObject::connect(timer,
+                     &QTimer::timeout,
+                     [c = lager::context<tick>{store}] { c.dispatch(tick{}); });
+    timer->start(initial_update_period);
+
+    QGame game{store};
+    watch(store, [=](auto&& old, auto&& state) {
+        if (!old.over && state.over) {
+            timer->stop();
+            return;
+        }
+
+        auto oldSpeed = int((old.snake.body.size() - initial_snake_size) / 5);
```

Comment:
- `old_snake`?
- C-style casts vs `static_cast`?

Reply:
Personally, I find `static_cast` way too verbose for casting between fundamentals. Note this isn't a C-style cast, but a functional-style cast. Essentially does the same, though. If you have a strong opinion, I can change it.